onnxruntime/core/optimizer/pad_fusion.cc

```diff
@@ -115,18 +146,44 @@ Status PadFusion::Apply(Graph& graph, Node& pad_node, RewriteRuleEffect& rule_ef
   }
 
   Node& child_node = *graph.GetNode(pad_node.OutputNodesBegin()->Index());
-  auto child_pads = child_node.GetMutableAttributes()["pads"].mutable_ints();
-  uint32_t child_pads_size = static_cast<uint32_t>(child_pads->size());
 
-  for (uint32_t pads_index = 2, child_index = 0; pads_index < pads_size / 2; pads_index++, child_index++) {
-    child_pads->Set(child_index, child_pads->Get(child_index) + pads_values[pads_index]);
-    uint32_t mirrored_child_index = child_index + (child_pads_size / 2);
-    uint32_t mirrored_pad_index = pads_index + (pads_size / 2);
-    child_pads->Set(mirrored_child_index, child_pads->Get(mirrored_child_index) + pads_values[mirrored_pad_index]);
+  if (child_node.OpType() == "Cast") {
+    if (pad_node.SinceVersion() >= 11) {
+      if (pad_node.InputDefs().size() > 2) {
+        auto* pad_constant_value_proto = graph_utils::GetConstantInitializer(graph, pad_node.InputDefs()[2]->Name());
```

Comment:
Why does the pad initializer data type need to be converted as the pad node is removed below?

Reply:
Yes, it is not required because the pad_constant_value needs to be zero for this fusion to work.